airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py

```diff
@@ -103,12 +114,14 @@ def __init__(
         self,
         *,
         dynamodb_table_name: str,
+        source_aws_conn_id: str = "aws_default",
```

Comment:
```suggestion
        source_aws_conn_id: str | None = "aws_default",
```

Reply:
On one side I agree that this typing would make more sense, however, the old implementation had `aws_conn_id` typed as `str` so people would just ignore they typing error messages when they would use `None` as a value? Contrary to the documentation, it only supported empty string before.